README.md
# IAMMOKZEEEE.github.io

Welcome to my coding portfolio website! This website is a showcase of my coding projects, blog posts, and a contact form to reach me.

## Description

### Sections

#### The Beginning
The header section of the website contains the main heading, as well as a navigation menu that allows you to quickly jump to other sections of the page.

#### A Peek Into My World
The About Me section provides a brief overview of who I am, my background, and my interests.

#### What I'm Working On
The What I'm Working On section showcases my current coding projects. It features 5 bootstrap cards that provide a brief summary of each project and a button that links to a separate project page. The project pages contain the title of the project, an in-depth description of the project, and a working web-app example if available.

#### The Time Capsule
The Time Capsule section is a searchable list of my previous coding projects. It includes a search bar and filters that allow you to sort the projects by code language. Each project has a button that links to a separate project page with more details.

#### Byte-Sized Insights
The Byte-Sized Insights section is a blog that features my latest thoughts on programming and technology. It is an RSS feed to a Jekyll page, with a home button that redirects back to the Byte-Sized Insights section of index.html.

#### Byte Buddy
The Byte Buddy section is a contact form where you can reach out to me. It includes four input fields: First Name, Last Name, Company, Email Address, and a text area for your message. All input fields are required except for Company. The form submission is handled by an un-static form manager.

### Offcanvas Sidebar Menu
The website includes an offcanvas sidebar menu that provides quick links to different sections of the website. The menu button is fixed in the top right-hand corner of the website and is signified by three horizontal circles.

### Home Button
All pages have a home button in the top-right corner with a purple background and a white house icon. This button allows you to quickly return to the corresponding section of index.html.

## Features
+ JavaScript Count-Up clock for how long I've been coding.
+ Bootstrap Sidebar Menu
+ Contact Form
+ Archive of previous projects and, where possible, ability to use them on-site.
+ Showcase current projects and, where possible, ability to use them on-site.

## License
This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details.

The MIT License is a permissive open source license that allows for free use, modification, and distribution of the code. This means you can use this project for any purpose, including commercial applications, and make any modifications you like. However, the license does come with some limitations and responsibilities, so please read the license terms carefully before using this code.

If you have any questions or concerns about the license or usage of this code, please feel free to contact me.


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## 1.1.1 (UNRELEASED)
### Changed
+ Archive List styling

## 1.1.0 (UNRELEASED)
### Added
+ Added list functionality on archive
+ Added CSS


## 1.0.0 (UNRELEASED)
### Added
+ New style.css
+ reset.css added
+ Re-introduced Bootstrap
+ Added jekyll blog
+ Added header to index.html
+ Added A Peek Into My Life to index.html
+ Added What I'm Working On to index.html
+ Added The Time Capsule to index.html
+ Added Byte Sized Insights to index.html
+ Added Byte Buddy to index.html
+ Added Bootstrap sidebar
+ Added Navigation to each section of index.html to sidebar
+ Added Bootstrap sidebar button
+ Added Bootstrap cards for What I'm Working On section


script/howLongIveBeenCoding.js
window.onload = function() {
    // Month Day, Year Hour:Minute:Second, id-of-element-container
    countUp("Mar 11, 2022 12:00:00", 'timer');
  };
  
  function countUp(countFrom, id) {
    countFrom = new Date(countFrom).getTime();
    let now = new Date(),
        countFrom = new Date(countFrom),
        timeDifference = (now - countFrom);
      
    let secondsInADay = 60 * 60 * 1000 * 24,
        secondsInAHour = 60 * 60 * 1000;
      
    days = Math.floor(timeDifference / (secondsInADay) * 1);
    hours = Math.floor((timeDifference % (secondsInADay)) / (secondsInAHour) * 1);
    mins = Math.floor(((timeDifference % (secondsInADay)) % (secondsInAHour)) / (60 * 1000) * 1);
    secs = Math.floor((((timeDifference % (secondsInADay)) % (secondsInAHour)) % (60 * 1000)) / 1000 * 1);
  
    document.getElementById('days')[0].innerHTML = days;
    document.getElementById('hours')[0].innerHTML = hours;
    document.getElementById('minutes')[0].innerHTML = mins;
    document.getElementById('seconds')[0].innerHTML = secs;
  
    clearTimeout(countUp.interval);
    countUp.interval = setTimeout(function(){ countUp(countFrom, id); }, 1000);
  }

script/script.js
/*
MIT License

Copyright (c) 2023 Charlie Marshall

Permission is hereby granted, free of charge, to any person obtaining a copy
of this software and associated documentation files (the "Software"), to deal
in the Software without restriction, including without limitation the rights
to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
copies of the Software, and to permit persons to whom the Software is
furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in
all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
THE SOFTWARE.
*/


//TIMER

window.onload = function() {
  countUpFromTime("Mar 11, 2022 12:00:00", 'timer');
};

function countUpFromTime(countFrom, id) {
  countFrom = new Date(countFrom).getTime();
  var now = new Date(),
      countFrom = new Date(countFrom),
      timeDifference = (now - countFrom);
    
  var secondsInADay = 60 * 60 * 1000 * 24,
      secondsInAHour = 60 * 60 * 1000;
    
  days = Math.floor(timeDifference / (secondsInADay) * 1);
  hours = Math.floor((timeDifference % (secondsInADay)) / (secondsInAHour) * 1);
  mins = Math.floor(((timeDifference % (secondsInADay)) % (secondsInAHour)) / (60 * 1000) * 1);
  secs = Math.floor((((timeDifference % (secondsInADay)) % (secondsInAHour)) % (60 * 1000)) / 1000 * 1);

  var idEl = document.getElementById(id);
  idEl.getElementsByClassName('days')[0].innerHTML = days;
  idEl.getElementsByClassName('hours')[0].innerHTML = hours;
  idEl.getElementsByClassName('minutes')[0].innerHTML = mins;
  idEl.getElementsByClassName('seconds')[0].innerHTML = secs;

  clearTimeout(countUpFromTime.interval);
  countUpFromTime.interval = setTimeout(function(){ countUpFromTime(countFrom, id); }, 1000);
}

//RSS FEED

$(document).ready(function() {
  $('#feed-container').FeedEk({
    FeedUrl: './blog/_site/feed.xml',
    MaxCount: 10,
    ShowDesc: true,
    ShowPubDate: true
  });
});

// Search function

function search() {
  // Declare variables
  var input, filter, li, i, txtValue, projects, row, col, a, h3;
  input = document.getElementById('searchBar');
  filter = input.value.toUpperCase();
  projects = document.getElementsByClassName("archive-project");

  // Loop through all list items, and hide those that don't match the search query
  for (i = 0; i < projects.length; i++) {
    row = projects[i].getElementsByClassName("row")[0];
    col = row.getElementsByClassName("li-title")[0];
    a = col.getElementsByTagName("a")[0];
    h3 = a.getElementsByTagName("h3")[0];

    txtValue = h3.textContent || h3.innerText;
    if (txtValue.toUpperCase().indexOf(filter) > -1) {
      projects[i].style.display = "";
    } else {
      projects[i].style.display = "none";
    }
  }
}


//List Sorting

function sortListDir() {
  var list, i, switching, b, shouldSwitch, dir, switchcount = 0;
  list = document.getElementById("id01");
  switching = true;
  // Set the sorting direction to ascending:
  dir = "asc";
  // Make a loop that will continue until no switching has been done:
  while (switching) {
    // Start by saying: no switching is done:
    switching = false;
    b = list.getElementsByTagName("LI");
    // Loop through all list-items:
    for (i = 0; i < (b.length - 1); i++) {
      // Start by saying there should be no switching:
      shouldSwitch = false;
      /* Check if the next item should switch place with the current item,
      based on the sorting direction (asc or desc): */
      if (dir == "asc") {
        if (b[i].innerHTML.toLowerCase() > b[i + 1].innerHTML.toLowerCase()) {
          /* If next item is alphabetically lower than current item,
          mark as a switch and break the loop: */
          shouldSwitch = true;
          break;
        }
      } else if (dir == "desc") {
        if (b[i].innerHTML.toLowerCase() < b[i + 1].innerHTML.toLowerCase()) {
          /* If next item is alphabetically higher than current item,
          mark as a switch and break the loop: */
          shouldSwitch= true;
          break;
        }
      }
    }
    if (shouldSwitch) {
      /* If a switch has been marked, make the switch
      and mark that a switch has been done: */
      b[i].parentNode.insertBefore(b[i + 1], b[i]);
      switching = true;
      // Each time a switch is done, increase switchcount by 1:
      switchcount ++;
    } else {
      /* If no switching has been done AND the direction is "asc",
      set the direction to "desc" and run the while loop again. */
      if (switchcount == 0 && dir == "asc") {
        dir = "desc";
        switching = true;
      }
    }
  }
}